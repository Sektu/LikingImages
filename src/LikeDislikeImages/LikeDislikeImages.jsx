import CustomizedImage from "../CustomizedImage/CustomizedImage";
import data from "../data"

const LikeDislikeImages = () => {
    const elem1 = data[0];
    return <>
        <CustomizedImage {...elem1} />
    </>;
}

export default LikeDislikeImages;