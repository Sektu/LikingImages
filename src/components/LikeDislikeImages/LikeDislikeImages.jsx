import CustomizedImage from "../CustomizedImage/CustomizedImage";
import UploadImage from "../UploadImage/UploadImage";
import { useSelector, useDispatch } from "react-redux";

const LikeDislikeImages = () => {  
    const images = useSelector((state) => state.images.data);
    
    return <>
        <UploadImage />
        {images.map((elem, iter) => {
            return <CustomizedImage key={iter} {...elem} />;
        })}
    </>;
}

export default LikeDislikeImages;