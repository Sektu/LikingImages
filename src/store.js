import { configureStore } from "@reduxjs/toolkit";
import likingReducer from "./slices/likingSlice";
import imagesReducer from "./slices/imagesSlice";

export default configureStore({
  reducer: {
    liking: likingReducer,
    images: imagesReducer,
  }
});
