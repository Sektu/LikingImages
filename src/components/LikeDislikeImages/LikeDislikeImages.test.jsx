import React from 'react';
import { render, cleanup } from '@testing-library/react';
import LikeDislikeImages from './LikeDislikeImages';
import store from "../../store";
import { Provider } from "react-redux";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Provider store={store}><LikeDislikeImages /></Provider>);
  expect(asFragment()).toMatchSnapshot();
});

it("check like dislike images", () => {
    const { getAllByRole } = render(<Provider store={store}><LikeDislikeImages /></Provider>);

    const headings = getAllByRole("heading");

    headings.forEach((h) => {
      expect(h).toBeInTheDocument();
      expect(h).toBeVisible();
      expect(h).toHaveClass('image-name');
    });

    expect(headings.length).toEqual(3);

    const imgs = getAllByRole("img");

    imgs.forEach((i) => {
      expect(i).toBeInTheDocument();
      expect(i).toBeVisible();
      expect(i).toHaveClass('image');
      expect(i).not.toHaveAttribute('alt', "");
      expect(i).not.toHaveAttribute('src', "");
    });

    expect(imgs.length).toEqual(3);

    const buttons = getAllByRole("button");

    buttons.forEach((b) => {
      expect(b).toBeInTheDocument();
      expect(b).toBeVisible();
      expect(b).toHaveClass('button');
      expect(b).toHaveClass('like-button');
    });

    expect(buttons.length).toEqual(3);
});
