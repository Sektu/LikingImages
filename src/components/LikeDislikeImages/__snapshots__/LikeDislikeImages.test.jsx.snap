// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <input
    accept="image/*"
    type="file"
  />
  <h5
    class="image-name"
  >
    calathea
  </h5>
  <img
    alt="calathea"
    class="image"
    src="https://i.pinimg.com/originals/7c/e5/4a/7ce54ab9433b1a133c4cbe9aa95b1f51.jpg"
  />
  <div>
    <button
      class="button like-button"
    >
      Like
    </button>
  </div>
  <h5
    class="image-name"
  >
    alocasia
  </h5>
  <img
    alt="alocasia"
    class="image"
    src="https://zielony-parapet.pl/18278-thickbox_default/alocasia-reginula-black-velvet-.jpg"
  />
  <div>
    <button
      class="button like-button"
    >
      Like
    </button>
  </div>
  <h5
    class="image-name"
  >
    epipremnum
  </h5>
  <img
    alt="epipremnum"
    class="image"
    src="https://a.allegroimg.com/s512/11f58a/45a1160c4f9dbedd26c5b83d9bb6/EPIPREMNUM-PINNATUM-MARBLE-QUEEN-SCINDAPSUS"
  />
  <div>
    <button
      class="button like-button"
    >
      Like
    </button>
  </div>
</DocumentFragment>
`;
